### PR TITLE
doc: Fix issues related to javadoc in java 8 #1104

### PR DESCRIPTION
**Pull request by surli**
Fix all errors encountered while launching ```mvn javadoc:javadoc``` on Spoon. Lot of warnings are still remaining but don't block the goal.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -241,6 +241,11 @@
       <artifactId>commons-io</artifactId>
       <version>1.4</version>
     </dependency>
+    <dependency>
```
Comment (tdurieux) [inline pom.xml]: it is a test dependency.
Please specify the scope of the dependency.

**Comment by tdurieux** [comment]
Maybe it is a good idea that spoon encodes and decodes the comments.
What do you think?

**Comment by surli** [comment]
> Maybe it is a good idea that spoon encodes and decodes the comments.

Encodes when building model and decodes when pretty-printing?
I'm not sure about the benefits of doing that, but still it should be in another issue/PR.

File: pom.xml (hunk 1 above)
Comment (surli) [inline pom.xml]: Done.

**Pull request merged**